src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+# class sonic_yang. A separate file is used to avoid a single large file.
+
+import yang as ly
+import re
+import pprint
+
+from json import dump, load, dumps, loads
+from xmltodict import parse
+from os import listdir, walk, path
+from os.path import isfile, join, splitext
+from glob import glob
+
+# class sonic_yang methods
+
+"""
+load all YANG models, create JSON of yang models
+"""
+def loadYangModel(self):
+
+    try:
+        yangDir = self.yang_dir
+        self.yangFiles = glob(yangDir +"/*.yang")
+        for file in self.yangFiles:
+            if (self.load_schema_module(file) == False):
+                return False
+
+        # keep only modules name in self.yangFiles
+        self.yangFiles = [f.split('/')[-1] for f in self.yangFiles]
+        self.yangFiles = [f.split('.')[0] for f in self.yangFiles]
+        print(self.yangFiles)
+
+        # load json for each yang model
+        self.loadJsonYangModel()
+        # create a map from config DB table to yang container
+        self.createDBTableToModuleMap()
+
+    except Exception as e:
+        print("Yang Models Load failed")
+        raise e
+
+    return True
+
+"""
+load JSON schema format from yang models
+"""
+def loadJsonYangModel(self):
+
+    for f in self.yangFiles:
+        m = self.ctx.get_module(f)
+        if m is not None:
+            xml = m.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+            self.yJson.append(parse(xml))
+
+    return
+
+"""
+Create a map from config DB tables to container in yang model
+"""
+def createDBTableToModuleMap(self):
+
+    for j in self.yJson:
+        # get module name
+        moduleName = j['module']['@name']
+        if "sonic-head" in moduleName or "sonic-common" in moduleName:
+            continue;
+        # get all top level container
+        topLevelContainer = j['module']['container']
+        if topLevelContainer is None:
+            raise Exception("topLevelContainer not found")
+
+        assert topLevelContainer['@name'] == moduleName
+
+        container = topLevelContainer['container']
+        # container is a list
+        if isinstance(container, list):
+            for c in container:
+                self.confDbYangMap[c['@name']] = {
+                    "module" : moduleName,
+                    "topLevelContainer": topLevelContainer['@name'],
+                    "container": c
+                    }
+        # container is a dict
+        else:
+            self.confDbYangMap[container['@name']] = {
+                "module" : moduleName,
+                "topLevelContainer": topLevelContainer['@name'],
+                "container": container
+                }
+    return
+
+"""
+Get module, topLevelContainer and json container for a config DB table
+"""
+def get_module_top_container(self, table):
+    cmap = self.confDbYangMap
+    m = cmap[table]['module']
+    t = cmap[table]['topLevelContainer']
+    c = cmap[table]['container']
+    return m, t, c
+
+"""
+Crop config as per yang models
```

Comment:
Can you add more details on what this function does?

Reply:
It crops from config only those TABLEs, for which yang models is provided. I can add this in Code.